test/cipherlist_test.c

```diff
@@ -155,8 +155,25 @@ static int test_default_cipherlist(SSL_CTX *ctx)
     if (ctx == NULL)
         return 0;
 
-    if (!TEST_ptr(ssl = SSL_new(ctx))
-            || !TEST_ptr(ciphers = SSL_get1_supported_ciphers(ssl)))
+    if (!TEST_ptr(ssl = SSL_new(ctx)))
+        goto err;
+
+#if defined(OPENSSL_NO_TLS1_2) && defined(OPENSSL_NO_TLS1_3)
+    /*
+     * When only TLS<1.2 is available and the security level is 0,
+     * SSL_get1_supported_ciphers(ssl) retuns NULL.
+     * That changed with commit 7bf2e4d7f0c7ae19b7a8c416910886a7171e9820.
+     * Not sure of that is a bug or a feature, especially since
+     * that does not happen with no-tls1_2 no-ec no-dh.
```

Comment:
With these options the tls-1.3 is enabled and there will be the tls-1.3 ciphersuites. Although they in a sense depend on ec and dh availability - this can be supplied by some provider later. So IMO no bug here.

Reply:
There is no way to provide such a provider in these tests, right?
These tests are intended to test our own crypto provider, right?

Reply:
Yes, but I thought that you're talking about SSL_get1_supported_ciphers(ssl) returning or not returning NULL in some conditions being a bug. I.E., I do not ask for any code changes just remove the sentence `Not sure ....` to make the comment less confusing/confused.

Reply:
I updated the comment to reflect your arguments.

Reply:
Yes, I didn't understand why SSL_get1_supported_ciphers does not return NULL in that case as well,
I mean, it returns NULL when the security level is 0, but I can change that later,
and if the provider does not provide ec or dh, what's the difference, we could return NULL now,
and change our mind, when the provider changes.